libpod/container_internal_linux.go

```diff
@@ -466,30 +466,53 @@ func (c *Container) setupSystemd(mounts []spec.Mount, g generate.Generator) erro
 		g.AddMount(tmpfsMnt)
 	}
 
-	// rootless containers have no write access to /sys/fs/cgroup, so don't
-	// add any mount into the container.
-	if !rootless.IsRootless() {
+	unified, err := cgroups.IsCgroup2UnifiedMode()
```

Comment:
This probably a question for Tom to learn something.
Is unified true only when running root?  Otherwise, do you still need a!rootless check at 475?

Reply:
no, it checks whether the system is using cgroups v2 unified hierarchy.  So it doesn't depend on the user running Podman.